File: slidingWindow/python/brute_force.py
def find_averages_of_subarrays(K, arr):
  result = []
  total_loop = 0
  for i in range(len(arr) - K+1):
    # find sum of next 'K' elements
    _sum = 0.0
    loop = i
    print("Loop starts") if total_loop == 0 else print(f"Another loop {loop} bouta start")
    for j in range(i, i + K):
      _sum += arr[j]
      total_loop +=1
      print( i, j, _sum )
    result.append(_sum/K)  # calculate average

  return result


def main():
  result = find_averages_of_subarrays(5, [1, 3, 2, 6, -1, 4, 1, 8, 2])
  print("Averages of subarrays of size K: " + str(result))
main()

File: slidingWindow/javascript/brute_force.js
// ==========================================================
// QUESTION:
// Given an array, find the average of all subarrays of ‘K’
// contiguous elements in it.
// ==========================================================
function find_averages_of_subarrays(K, arr) {
  const result = [];
  let totalLoops = 0;
  for (let i = 0; i < arr.length - K + 1; i++) {
    //  i < arr.length - K + 1: loop through arrays
    //  and dont run out of window-rail-line for the window-size(k)
    let sum = 0.0;
    //  find sum of next 'K' elements
    let loop = i;
    totalLoops === 0
      ? console.log("Loop starts")
      : console.log(`Another loop ${loop} bouta start`);

    for (let j = i; j < i + K; j++) {
      // j < i + K: sum from the new window-tail to the window size k
      sum += arr[j];
      totalLoops++;
      console.log({
        i,
        j,
        sum,
        nested_con: i + K,
      });
    }
    // now that we have sum of the subrrray
    // we can push it into our result array
    result.push(sum / K);
  }
  return console.log({
    TotalNumOfLoops: totalLoops,
    TimeComplexity: "O(N*K)",
    result,
  });
}
const result = find_averages_of_subarrays(5, [1, 3, 2, 6, -1, 4, 1, 8, 2]);
